tests/extractors/extract_from_files_test.py

```diff
@@ -0,0 +1,57 @@
+import contextlib
+import io
+import typing
+from concurrent.futures import Executor
+from pathlib import Path
+from unittest import mock
+
+import pytest
+from aiohttp import ClientSession
+
+from features.extract_from_files import ExtractFromFiles
+from tests.integration.features_integration_test import mock_website_data
+
+
+@contextlib.contextmanager
+def file_download_mock():
+    """Patch the ExtractFromFiles download method to instead load the files from the test resources."""
+
+    async def download_mock(self, file_url: str, session: ClientSession) -> typing.BinaryIO:
+        if "arbeitsblatt_analog_losung.docx" in file_url:
```

Comment:
Why do you trat them differently? as far as I can see it, both are similar, it is just the Path, that changes.
And I do not understand the else clause - this is a test, but the else clause indicates, that there might be a chance of an unwanted test? I am confused.

Reply:
the mock only supports those two input values. so it must fail if some other file content is requested, hence the exception.

The two if else are just a "lazy" approach to map from the file_url argument to the path that needs to be opened.

Reply:
oh yeah, i simplified this quite a bit now - thanks :)